src/frequenz/sdk/actor/power_distributing/_battery_status.py

```diff
@@ -352,6 +354,26 @@ def _get_current_status(self) -> Status:
 
         return Status.WORKING
 
+    def _is_capacity_present(self, msg: BatteryData) -> bool:
+        """Check whether the battery capacity is NaN or not.
+
+        If battery capacity is missing, then we can't work with it.
+
+        Args:
+            msg: battery message
+
+        Returns:
+            True if battery capacity is missing, false otherwise.
+        """
+        if math.isnan(msg.capacity):
+            if self._last_status == Status.WORKING:
+                _logger.warning(
+                    "Battery %d capacity is NaN",
+                    msg.component_id,
+                )
```

Comment:
How often will this message be produced if a battery capacity is `NaN` for a prolonged period? Only once? Once per resampling period?

Not for this PR, but more like a mental note: the more I think about alerts, the more I'm convinced in the future we need to use counters instead of ERROR/WARNINGs to produce alerts. In this case for example, we probably want to set an alert for `broken_battery_count > 0` and receive it only once, and have it in alerting state until it goes back to `broken_battery_count == 0`.

FYI @matthias-wende-frequenz @thomas-nicolai-frequenz

Reply:
I'm not sure if that should be a worry of the SDK of how the alerting will work for now. Feels too much to address right now. Though, it doesn't mean that I disagree but it feels to early to address that problem.

Reply:
> How often will this message be produced if a battery capacity is NaN

Only if latest battery status was WORKING.
If capacity is NaN, then the status will change to NOT_WORKING and the log won't be printed.

Reply:
> I'm not sure if that should be a worry of the SDK of how the alerting will work for now. Feels too much to address right now. Though, it doesn't mean that I disagree but it feels to early to address that problem.

Yeah, definitely, thinking about the mid-long term here, didn't mean now. But sometimes when talking about this in the air is not easy to come up with good examples, so I just wanted to point this out.